lab/torch/random.py

```diff
@@ -22,17 +22,25 @@ def global_random_state(_: TorchDType):
     else:
         parts = B.ActiveDevice.active_name.lower().split(":")
 
-        if len(parts) == 0 or parts[0] not in {"cuda", "gpu"}:
+        if len(parts) == 0 or parts[0] not in {"cuda", "gpu", "mps"}:
             raise RuntimeError(f'Unknown active device "{B.ActiveDevice.active_name}".')
 
-        # Ensure that the generators are available.
-        if len(torch.cuda.default_generators) == 0:
-            torch.cuda.init()
+        if parts[0] == "mps":
+            import torch.mps
 
-        if len(parts) == 1:
-            return torch.cuda.default_generators[0]
+            if torch.mps._default_mps_generator:
+                return torch.mps._default_mps_generator
+            else:
+                return torch.mps._get_default_mps_generator()
```

Comment:
```suggestion
            return mps._get_default_mps_generator()
```

From reading the soruce of `torch.mps`, I believe this will always get the default generator, so this might be simpler.